web/client/components/data/featuregrid/EmptyRowsView.jsx

```diff
@@ -6,23 +6,24 @@
  * LICENSE file in the root directory of this source tree.
  */
 import React from 'react';
-
 import PropTypes from 'prop-types';
 import Message from '../../I18N/Message';
+import WidgetEmptyMessage from '../../widgets/widget/WidgetEmptyMessage';
+
 const nodataStyle = {
     width: "100%",
     height: "100%",
     textAlign: "center",
     verticalAlign: "center"
 };
+
 class EmptyRowsView extends React.PureComponent {
     static propTypes = {
         loading: PropTypes.bool
     }
     render() {
         return this.props.loading
-            ? (<div style={nodataStyle}><Message msgId="loading" /></div>) :
-            <div style={nodataStyle}><Message msgId="featuregrid.noFeaturesAvailable" /></div>;
+            ? (<div style={nodataStyle}><Message msgId="loading" /></div>) : (<WidgetEmptyMessage messageId="featuregrid.noFeaturesAvailable" glyph="features-grid"/>);
```

Comment:
Instead of modify this component used inside different grid we could give the possibility to customize the node message :

```js
class EmptyRowsView extends React.PureComponent {
    static propTypes = {
        loading: PropTypes.bool
    }
    render() {
        return this.props.loading
            ? (<div style={nodataStyle}><Message msgId="loading" /></div>) :
            this.props.children
                ? this.props.children
                : <div style={nodataStyle}><Message msgId="featuregrid.noFeaturesAvailable" /></div>;
    }
}
```

then we could pass the WidgetEmptyMessage only for the TableWidgets

https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/widgets/widget/TableWidget.jsx#L70

eg:

```js
// ...
emptyRowsView={() =>
    <EmptyRowsView loading={loading}>
        <WidgetEmptyMessage messageId="featuregrid.noFeaturesAvailable" glyph="features-grid"/>
    </EmptyRowsView>}
// ...
```

in this way the WidgetEmptyMessage is related only to the Widgets components.
If you think the this.props.children is not enough explicit you could use a props to pass the custom message node.

Reply:
this makes it a better implementation. Using props.children is enough